src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,199 @@
+                {
+                    String[] s = _https.UserInfo.Split(':', 2);
+                    _httpsCred = new NetworkCredential(s[0], s[1]);
+                }
+            }
+
+            string value = Environment.GetEnvironmentVariable("no_proxy");
+            if (!String.IsNullOrWhiteSpace(value)) 
+            {
+                _bypass = value.Split(',');
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. 
+        /// </summary>
+        private Uri GetUriFromString(String value)
+        {
+            if (!value.StartsWith("http://", StringComparison.OrdinalIgnoreCase) &&
+                !value.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+            {
+                // Default to HTTP
+                value = "http://" + value;
+            }
+            Uri uri = new Uri(value);
+
+            if (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps)
+            {
+                throw new NotSupportedException("Only HTTP and HTTPS protocols are supported for proxy.");
+            }
+            return uri;
+        }
+
+        /// <summary>
+        /// This function returns true if given Host mach bypss list.
+        /// Note, that the list is common for http and https.
+        /// </summary>
+        private bool IsMatchInBypassList(Uri input)
+        {
+            if (_bypass != null)
+            {
+                foreach (string s in _bypass)
+                {
+                    if (s[0] == '.')
+                    {
+                        // This should match either domain it self or any subdomain or host
+                        // .foo.com will match foo.com it self or *.foo.com
+                        if ((s.Length - 1) == input.Host.Length &&
+                            String.Compare(s, 1, input.Host, 0, input.Host.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            return true;
+                        }
+                        else if (input.Host.EndsWith(s, StringComparison.OrdinalIgnoreCase))
+                        {
+                                return true;
+                        }
+
+                    }
+                    else
+                    {
+                        if (String.Compare(s, input.Host, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            return true;
+                        }
+                    }
+
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the proxy URI. (iWebProxy interface)
+        /// </summary>
+        public Uri GetProxy(Uri uri)
+        {
+            return uri.Scheme == "http" ? _http : _https;
```

Comment:
I see that in `GetCredentials` below, `Uri.UriSchemeHttp` is used. Should that be used here instead of `"http"`?